docs/configuring-playbook-bot-matrix-registration-bot.md

```diff
@@ -0,0 +1,71 @@
+# Setting up matrix-registration-bot (optional)
+
+The playbook can install and configure [matrix-registration-bot](https://github.com/moanos/matrix-registration-bot) for you.
+
+The bot allows you to easily **create and manage registration tokens**. It can be used for an invitation-based server,
+where you invite someone by sending them a registration token. They can register as normal but have to provide a valid 
+registration token in a final step  of the registration.
+
+See the project's [documentation](https://github.com/moan0s/matrix-registration-bot#supported-commands) to learn what it
+does and why it might be useful to you.
+
+
+## Registering the bot user
+
+By default, the playbook will set use the bot with a username like this: `@bot.matrix-registration-bot:DOMAIN`.
+
+(to use a different username, adjust the `matrix_bot_matrix_registration_bot_matrix_user_id_localpart` variable).
+
+You **need to register the bot user manually** before setting up the bot. You can use the playbook to [register a new user](registering-users.md):
+
+```
+ansible-playbook -i inventory/hosts setup.yml --extra-vars='username=bot.matrix-registration-bot password=PASSWORD_FOR_THE_BOT admin=no' --tags=register-user
+```
+
+Choose a strong password for the bot. You can generate a good password with a command like this: `pwgen -s 64 1`.
+
+## Obtaining an admin API token
+
+In order to use the bot you need to add an admin API token to the configuration. Log into Element/Schildichat with an
+administrator account and navigate to `Settings->Help&About` and scroll to the bottom. You can expand "Access token" to copy it.
+
+![Obatining an admin access token with Element](assets/obtain_admin_access_token_element.png)
+
+## Adjusting the playbook configuration
+
+Add the following configuration to your `inventory/host_vars/matrix.DOMAIN/vars.yml` file:
+
+```yaml
+matrix_bot_matrix_registration_bot_enabled: true
+# Adjust this to whatever password you chose when registering the bot user
+matrix_bot_matrix_registration_bot_matrix_user_password: "PASSWORD_FOR_THE_BOT"
+# Token obtained via an administrator account
+matrix_bot_matrix_registration_bot_matrix_admin_token: "syt_bW9hbm9z_XXXXXXXXXXXXXr_2kuzbE"
+
+# Enables registration
+matrix_synapse_enable_registration: true
+
+# Restrict registration to users with a token
+matrix_synapse_configuration_extension_yaml: |
+  registration_requires_token: true # For token based registration
```

Comment:
Let's add a dedicated variable (`matrix_synapse_registration_requires_token: false` by default) for this. We can then simplify by not having to use `matrix_synapse_configuration_extension_yaml`.

Reply:
This was already done: https://github.com/spantaleev/matrix-docker-ansible-deploy/commit/f6cb59116b93bc826fdb2a7b8999046eaa08e0d6
I will be using it now (didn't exist when I looked for it last time)